### PR TITLE
[Validator] Allow skipping import validation (complete)

### PR DESCRIPTION
**Pull request by dnkoutso**
@segiddins this completes https://github.com/CocoaPods/CocoaPods/pull/5677

* I dont think I could have forced pushed to your branch, let me know if you want me to do that instead
* I rebased to latest and fixed conflicts
* Reverted your change for the `import_statement`
* Added a spec and updated existing ones to pass
* Added CHANGELOG entry

The committer name is still your name, because you did 99.9% of this.

### CODE REVIEW

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -425,6 +427,8 @@ def add_app_project_import
       add_swift_version(app_target)
       add_xctest(app_target) if @installer.pod_targets.any? { |pt| pt.spec_consumers.any? { |c| c.frameworks.include?('XCTest') } }
       app_project.save
+      Xcodeproj::XCScheme.share_scheme(app_project.path, 'App')
+      # Xcodeproj::XCScheme.share_scheme(@installer.pods_project.path, pod_target.label)
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: @segiddins adding this causes the tests to fail. I think `pods_project` needs to be stubbed but the tests are a bit complicated to figure out. Let me know if something quick comes up for this.

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -764,7 +768,12 @@ def parse_xcodebuild_output(output)
     #
     def xcodebuild
       require 'fourflusher'
-      command = ['clean', 'build', '-workspace', File.join(validation_dir, 'App.xcworkspace'), '-scheme', 'App', '-configuration', 'Release']
+      scheme = if skip_import_validation?
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: rubocop styling......

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Add `--skip-import-validation` to skip linking a pod during lint.  
+  [Samuel Giddins](https://github.com/segiddins) & [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
```
Comment (segiddins) [inline CHANGELOG.md]: no &, each author on a separate line

Comment (dnkoutso) [inline CHANGELOG.md]: fixed!

**Comment by dnkoutso** [comment]
Locally tested it in a pod with the flag on and worked.

**Comment by dnkoutso** [comment]
@DanToml can this land now that 1.2.0 is out?

**Comment by segiddins** [comment]
Beat me to it :(

**Comment by igagis** [comment]
I which release is this available?

**Pull request merged**